docs/developer_guide/versioning.md

```diff
@@ -0,0 +1,64 @@
+
+**What's Not a Breaking Change?**
+
+Here are some changes you can expect without worrying about breaking your code:
+
+- **Bug Fixes:** We continuously fix bugs to improve MedModels' performance and stability. These fixes won't require code modifications.
+- **Internal Restructuring:** Sometimes, we might reorganize internal code for better efficiency. This won't affect how you use MedModels functionalities.
+- **Optional Enhancements:** We might introduce new optional parameters to existing methods, providing more flexibility without breaking existing code.
+
+**Keeping You Informed**
+
+We prioritize clear communication. Whenever possible, we'll introduce deprecation warnings for features planned for removal. We'll also provide detailed changelogs and migration guides to help you adapt your code seamlessly to any breaking changes.
+
+By understanding these guidelines, you can stay informed about changes in MedModels and ensure your code continues to function effectively with the latest advancements.
+
+## Deprecation Warnings: Saying Goodbye (But Not Without Notice)
+
+Whenever possible, we give functionalities a heads-up before retirement! This means you'll get a deprecation warning if we plan to remove a feature. For example, imagine a function gets a promotion and a new name. We'll keep the old function around for a while, but it will politely warn you to use the new, improved version instead.
+
+**Some Changes Require Bigger Announcements:**
+
+However, not all improvements are as straightforward. Occasionally, major advancements to our internal algorithms might necessitate broader changes to how you interact with MedModels. In these cases, individual functionalities might not receive specific deprecation warnings. But don't worry, we'll still keep you informed! We'll provide detailed explanations in the changelog and a handy migration guide to help you smoothly transition your code to the new and improved MedModels.
+
+## Giving You Time to Adjust: The Deprecation Window
+
+Once a functionality is marked for retirement (deprecation), we typically give it a two-version grace period before it's officially removed. Here's how it works:
+
+- **Before Version 1.0.0:** Imagine a function gets marked for retirement in version 0.18.3. It will stick around for two minor version bumps, finally saying goodbye in version 0.20.0.
+- **After Version 1.0.0:** Once we reach version 1.0.0, things change slightly. A function deprecated in version 1.2.3 will be removed two major versions later, in version 3.0.0.
+
+This system provides you with ample time to adjust your code. As minor versions typically release every three months, you'll generally have three to six months to prepare for any upcoming breaking changes. That means you can update your code at your own pace, ensuring a smooth transition with the latest MedModels advancements.
```

Comment:
There are some biiiiiiiiiiiiiiig claims in this file. I'm not sure we're quite ready to make them yet. I'd suggest to leave them out for now and add them later when we actually have a stable package